src/SIL.XForge.Scripture/ClientApp/src/app/shared/audio-player.ts

```diff
@@ -0,0 +1,125 @@
+import { BehaviorSubject } from 'rxjs';
+import { formatFileSource, isLocalBlobUrl } from 'xforge-common/file.service';
+import { FileType } from 'xforge-common/models/file-offline-data';
+import { PwaService } from 'xforge-common/pwa.service';
+import { SubscriptionDisposable } from 'xforge-common/subscription-disposable';
+
+export enum AudioStatus {
+  Init = 'audio_initialized',
+  Available = 'audio_available',
+  Unavailable = 'audio_cannot_be_accessed',
+  LocalNotAvailable = 'audio_cannot_be_previewed',
+  Offline = 'audio_cannot_be_played'
+}
+
+// See explanatory comment where this number is used
+const ARBITRARILY_LARGE_NUMBER = 1e10;
+
+export class AudioPlayer extends SubscriptionDisposable {
+  private static lastPlayedAudio: HTMLAudioElement;
+  private audio: HTMLAudioElement = new Audio();
+  private audioDataLoaded: boolean = false;
+
+  status$: BehaviorSubject<AudioStatus> = new BehaviorSubject<AudioStatus>(AudioStatus.Init);
+
+  constructor(source: string, private readonly pwaService: PwaService) {
+    super();
+    this.audio.addEventListener('loadeddata', () => {
+      this.currentTime = 0;
+      this.audioDataLoaded = true;
+      this.status$.next(AudioStatus.Available);
+    });
+
+    this.audio.addEventListener('timeupdate', () => {});
```

Comment:
This is required to cause consuming components to update seek/currentTime values at the expected interval. Before, they were only updating once every few seconds. It seems less than ideal, yet better than forcing all components to listen to some new event and trigger their own UI updates.